File: lcs.py
import sys

def lcs_arr(a, b):
    m = max(len(a) + 1, len(b) + 1)
    L = [[0 for x in xrange(m)] for x in xrange(m)]
    amax = len(a)
    bmax = len(b)
    for i in range(amax, -1, -1):
        for j in range(bmax, -1, -1):
            if i == amax or j == bmax: continue
            elif a[i] == b[j]: L[i][j] = 1 + L[i+1][j+1]
            else: L[i][j] = max(L[i+1][j], L[i][j+1])
            
    return L

def lcs(a, b):
    L = lcs_arr(a, b)
    s = ""
    i = 0
    j = 0
    m = len(a)
    n = len(b)
    while i < m and j < n:
        if a[i] == b[j]:
            s = s + a[i]
            i = i + 1
            j = j + 1
        elif (L[i+1][j] >= L[i][j+1]): i = i + 1
        else: j = j + 1

    return s

f = open(sys.argv[1])
for l in f:
    l = l.strip()
    if l:
        [a, b] = l.split(';')
        s = lcs(a, b)
        print s

File: lcs.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int max(int x, int y){
  if (x == y) return x;
  else if (x > y) return x;
  else return y;
}

int **lcs_arr(const char* a, const char *b){
  printf("comparing: %s with %s\n", a, b);
  int m = strlen(a);
  int n = strlen(b);
  int **L = calloc((m+1) * (n+1), sizeof(int));
  if (L == NULL) {
    printf("WTF calloc!\n");
  }
  for(int i=m; i>= 0; i--){
    for(int j=n; j >= 0; j--){
      printf("m:%d,n:%d,i:%d,j:%d\n", m, n, i, j);
      if (i == m || j == n) continue;
      else if (a[i] == b[j]) L[i][j] = 1 + L[i+1][j+1];
      else L[i][j] = max(L[i+1][j], L[i][j+1]);
    }
  }
  return L;
}
 
const char* lcs(const char *a, const char *b){
  return NULL;
}

int main(int argc, char **argv){
  size_t SIZE = 1024;
  char b[SIZE];
  char *token, *token1;
  FILE *f;

  f = fopen(argv[1], "r");
  while ( (fgets(b, SIZE , f) != NULL) ){
    token = strtok(b, ";");
    token1 = token;
    token = strtok(NULL, ";");
    int **arr = lcs_arr(token1, token);

  }
  fclose(f);
}
